### PR TITLE
Fix some SpotBugs violations

### PR DESCRIPTION
**Pull request by basil**
Triages/fixes some low-hanging fruit in the SpotBugs violations file. CC @StefanSpieker 

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/PluginManager.java
Comment (basil) [inline core/src/main/java/hudson/PluginManager.java:277]: Overly defensive programming. This can never happen.

File: core/src/main/java/hudson/PluginManager.java
Comment (basil) [inline core/src/main/java/hudson/PluginManager.java:915]: As of #5931 this is a `ConcurrentMap`, so no synchronization is necessary.

File: core/src/main/java/hudson/model/Run.java
Comment (basil) [inline core/src/main/java/hudson/model/Run.java:1916]: This is always non-null, but the static analyzer can't figure it out. Adding this call helps the static analyzer and makes the code fail fast if a future refactoring breaks the existing invariant.

File: core/src/main/java/jenkins/FilePathFilterAggregator.java
Comment (basil) [inline core/src/main/java/jenkins/FilePathFilterAggregator.java:33]: `Double.compare` returns an `int`.

File: core/src/main/java/jenkins/model/Jenkins.java
Comment (basil) [inline core/src/main/java/jenkins/model/Jenkins.java:2439]: This can never be null because `ExtensionList#getInstance` throws `IllegalStateException` if the class cannot be found.

File: core/src/main/java/jenkins/model/Jenkins.java
Comment (basil) [inline core/src/main/java/jenkins/model/Jenkins.java:2451]: This can never be null because `ExtensionList#getInstance` throws `IllegalStateException` if the class cannot be found.

File: core/src/main/java/jenkins/model/lazy/LazyBuildMixIn.java
Comment (basil) [inline core/src/main/java/jenkins/model/lazy/LazyBuildMixIn.java:366]: A more direct way of expressing the same concept that also aids the static analyzer.

File: core/src/main/java/jenkins/security/s2m/AdminWhitelistRule.java
Comment (basil) [inline core/src/main/java/jenkins/security/s2m/AdminWhitelistRule.java:77]: Seems like a legitimate (though minor) resource leak.

File: core/src/main/java/hudson/model/queue/CauseOfBlockage.java
Comment (basil) [inline core/src/main/java/hudson/model/queue/CauseOfBlockage.java:49]: A more direct way of expressing the same concept that also aids the static analyzer.

File: core/src/main/java/hudson/security/AuthenticationProcessingFilter2.java
Comment (daniel-beck) [inline core/src/main/java/hudson/security/AuthenticationProcessingFilter2.java:64]: Something for later would be to remove the option if (new) telemetry reveals this is never used. We chose to add this for safety in a security update just in case `SessionFixationProtectionStrategy` failed for some realms.

**Review by daniel-beck — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
